packages/fxa-auth-server/lib/routes/subscriptions/stripe-webhook.ts

```diff
@@ -425,6 +432,64 @@ export class StripeWebhookHandler extends StripeHandler {
     await this.updateCustomer(uid, email);
   }
 
+  /**
+   * Validate plan metadata on update
+   */
+  async handlePlanUpdatedEvent(request: AuthRequest, event: Stripe.Event) {
+    const plan = event.data.object as Stripe.Plan;
+    if (typeof plan.product !== 'string') return;
+    const product = await this.stripeHelper.fetchProductById(plan.product);
+    if (!product) {
```

Comment:
I don't think this condition is possible.  If `stripe.products.retrieve` returns a product, then `abbrevProductFromStripeProduct` would return an object.  If `stripe.products.retrieve` fails/throws, then the `abbrevProductFromStripeProduct` return statement won't be reached anyway.

In order for this to make sense, we'd need to try-catch in `fetchProductById` and return a null in the catch.